### PR TITLE
SONAR-7920 Rewrite Links project page

### PR DESCRIPTION
**Pull request by stas-vilchik**

### CODE REVIEW

File: server/sonar-web/src/main/js/apps/project-admin/links/views/DeletionModal.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+ */
+import ModalForm from '../../../../components/common/modal-form';
+import Template from './DeletionModalTemplate.hbs';
+import { deleteLink } from '../../../../api/projectLinks';
+
+export default ModalForm.extend({
+  template: Template,
+
+  onFormSubmit () {
+    ModalForm.prototype.onFormSubmit.apply(this, arguments);
```
Comment (SonarTech) [inline server/sonar-web/src/main/js/apps/project-admin/links/views/DeletionModal.js]: MAJOR Use the rest syntax to declare this function's arguments. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=javascript%3AS3513)

File: server/sonar-web/src/main/js/apps/project-admin/links/views/CreationModal.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+import ModalForm from '../../../../components/common/modal-form';
+import Template from './CreationModalTemplate.hbs';
+
+export default ModalForm.extend({
+  template: Template,
+
+  onFormSubmit () {
+    ModalForm.prototype.onFormSubmit.apply(this, arguments);
```
Comment (SonarTech) [inline server/sonar-web/src/main/js/apps/project-admin/links/views/CreationModal.js]: MAJOR Use the rest syntax to declare this function's arguments. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=javascript%3AS3513)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues
* MAJOR 2 major

Watch the comments in this conversation to review them.

File: it/it-tests/src/test/java/it/projectAdministration/ProjectLinksPageTest.java [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+  private CreateWsResponse createCustomLink() {
+    return wsClient.projectLinks().create(new CreateWsRequest()
+      .setProjectKey("sample")
+      .setName("Custom")
+      .setUrl("http://example.org/custom"));
+  }
+
+  private void deleteLink(long id) {
+    try {
+      wsClient.projectLinks().delete(new DeleteWsRequest().setId(id));
```
Comment (teryk) [inline it/it-tests/src/test/java/it/projectAdministration/ProjectLinksPageTest.java]: I don't think you want it to fail silently. Eventually use `throw Throwables.propage(e)`. It's in each test that you want to handle it properly

**Comment by teryk** [comment]
small feedback and LGTM

**Pull request merged**